frontend/common/src/components/form/Select/Select.tsx

```diff
@@ -93,15 +150,37 @@ const Select: React.FunctionComponent<SelectProps> = ({
               {nullSelection}
             </option>
           )}
-          {options.map((option) => (
-            <option
-              aria-label={option.ariaLabel}
-              key={option.value}
-              value={option.value}
-            >
-              {option.label}
-            </option>
-          ))}
+          {optionsModified.map((option) =>
+            Object.prototype.hasOwnProperty.call(option, "options") ? (
+              <optgroup
+                key={`optgroup${option.label}`}
+                label={
+                  isNull(option.label) ||
+                  isNumber(option.label) ||
+                  isBoolean(option.label) ||
+                  isUndefined(option.label)
+                    ? ""
+                    : option.label.toString()
+                }
```

Comment:
Would ` label={option.label?.toString() || ""}` get to the same end result? I think the `?` will omit `null`, `undefined` and `false`. Though it would result in "true" being printed 🤔

Reply:
If this works, try `label={option.label?.toString() ?? ""}` instead. We almost always want to use [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#assigning_a_default_value_to_a_variable) instead of logical or for setting default values. That should avoid accidentally turning the expression into a boolean 😉 . Also it allows for falsey values like `0`.

Reply:
Makes sense to me! Though, converting to a string would make `0` and `false` truthy 😜 

![Screenshot 2023-02-17 105842](https://user-images.githubusercontent.com/4127998/219703143-ebfacc7d-cb69-4615-a888-a13f55901da3.png)

Reply:
This was another one that TypeScript was complaining about it and I ended up with the mess, sorry. "Resolved" in [Refactor label condition](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5729/commits/3748982f345014d205e69c490d494c8ab60f28a2).